File: src/app/models/question.ts
import { Answer } from './answer';
import { Category } from './category';
import { CategoryQuestion } from './categoryquestion';
import { environment } from '../../environments/environment.prod';

export class Question {
    constructor() {
        this.id = environment.emptyGuid;
    }
    id: string;
    title: string;
    answers: Answer[];
    creationDate: Date;
    deletionDate: Date;
    isDeleted: boolean;
    categoryList: CategoryQuestion[];
}

File: API/Models/Question.cs
using System.Collections;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System;

namespace QuizAppApi.Models
{
    public class Question
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public Guid Id { get; set; }
        public string Title { get; set; }
        public List<Answer> Answers { get; set; }
        public DateTime? CreationDate { get; set; }
        public DateTime? DeletionDate { get; set; }
        public bool IsDeleted { get; set; }
        public List<CategoryQuestion> CategoryList { get; set; }
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public Guid CorrectAnswerId { get; set; }
    }
}
